src/main/java/com/avispl/dal/communicator/polycom/videoos/PolycomVideoOS.java

```diff
@@ -0,0 +1,1046 @@
+                if (null != conferenceId) {
+                    String remoteAddress = getJsonProperty(meetingInfo, "address", String.class);
+                    if (!StringUtils.isNullOrEmpty(remoteAddress) && remoteAddress.trim().equals(dialDevice.getDialString().trim())) {
+                        return conferenceId;
+                    }
+                }
+            }
+            Thread.sleep(1000);
+        }
+        return null;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * Locking is necessary because statistics is still being collected there and if the device is in the call
+     * and collecting some information available while in the call only - it'll end up with a 404 error.
+     * If there's a conferenceId available - only one conference is removed, otherwise - the method
+     * iterates through all of the available conferences and removes them.
+     */
+    @Override
+    public void hangup(String conferenceId) throws Exception {
+        if (logger.isDebugEnabled()) {
+            logger.debug("Hangup string received: " + conferenceId);
+        }
+        checkAndInitialize();
+        controlOperationsLock.lock();
+        try {
+            if (!StringUtils.isNullOrEmpty(conferenceId)) {
+                doDelete(String.format(CONFERENCES, conferenceId));
+            } else {
+                ArrayNode conferenceCall = listConferenceCalls();
+                for (JsonNode node : conferenceCall) {
+                    Integer id = getJsonProperty(node, "id", Integer.class);
+                    doDelete(String.format(CONFERENCES, id));
+                }
+            }
+        } finally {
+            controlOperationsLock.unlock();
+        }
+    }
+
+    @Override
+    public CallStatus retrieveCallStatus(String conferenceId) throws Exception {
+        if (logger.isDebugEnabled()) {
+            logger.debug("Retrieving call status with string: " + conferenceId);
+        }
+        checkAndInitialize();
+        controlOperationsLock.lock();
+        try {
+            if(!StringUtils.isNullOrEmpty(conferenceId)){
+                JsonNode response = doGet(String.format(CONFERENCES, conferenceId), JsonNode.class);
+                if (response == null) {
+                    return generateCallStatus(conferenceId, CallStatus.CallStatusState.Disconnected);
+                }
+                Boolean conferenceIsActive = getJsonProperty(response, "isActive", Boolean.class);
+                if (conferenceIsActive != null && conferenceIsActive) {
+                    return generateCallStatus(getJsonProperty(response, "id", String.class), CallStatus.CallStatusState.Connected);
+                }
+            } else {
+                ArrayNode conferences = listConferenceCalls();
+                if(conferences != null && conferences.size() > 0){
+                    return generateCallStatus(getJsonProperty(conferences.get(0), "id", String.class), CallStatus.CallStatusState.Connected);

```

Comment:
If you can call this logic as obvious - I cannot, sorry. So there are might be 10 conferences there, the code will build status based on just one of them (randomly? Are they ordered somehow? And are they really conferences? The local variable is conferences, the method is listConferenceCalls, they might be conferences, or calls, not one except you knows that)